### PR TITLE
Created a parent `extract` function

### LINKED ISSUE #49: Create a parent `extract` function
An `extract` function should be added that looks at the type or `instanceof` of the first argument and calls one of the appropriate child/type-specific extract functions, e.g. `extract_from_file`, `extract_from_files`, `extract_from_bytes`, `extract_from_url`, etc.

### PR DESCRIPTION
**Pull request by gabb222**
Fixes #49

### CODE REVIEW

**Comment by gabb222** [comment]
@Theia-Scientific/developers ready to review

File: tspng/extraction.py [hunk 1]
```diff
@@ -23,6 +23,31 @@ def _open_image(file_or_bytes: Union[str, io.BytesIO], mime_type: str=MIME_TYPE)
...
+    '''
+    #call appropriate function
+    if os.path.exists(file_bytes_or_files):
+        return extract_from_file(file_bytes_or_files, MIME_TYPE)
+    elif isinstance(file_bytes_or_files, io.BytesIO):
+        return extract_from_bytes(file_bytes_or_files, MIME_TYPE)
+    elif isinstance(file_bytes_or_files, list):
+        return extract_from_files(file_bytes_or_files, MIME_TYPE)
+    else:
+        raise TypeError(f"{file_bytes_or_files} is not a BytesIO object, file, or list of file.")
```
Comment (volks73) [inline tspng/extraction.py:49]: The "...list of file" should be "...list of files".

**Review by volks73 — APPROVED** [review]
This looks really good. While it may feel redundant, it is good. I like it.

One small possible revision, non-blocking.

File: tspng/extraction.py (hunk 1 above)
Comment (gabb222) [reply tspng/extraction.py]: Change made as requested

**Pull request merged**